### PR TITLE
Code refactoring

### PR DESCRIPTION
**Pull request by krmahadevan**
* Removed some deprecated methods
* Replaced verbose implementations with streams
* Removed redundant methods

### CODE REVIEW

**Comment by krmahadevan** [comment]
ping @juherr

File: src/main/java/org/testng/internal/GroupsHelper.java [hunk 1]
```diff
@@ -32,12 +33,8 @@ private GroupsHelper() {
    */
   public static Map<String, String> createGroups(
       Map<String, List<String>> metaGroups, List<String> groups) {
-    Map<String, String> result = Maps.newHashMap();
-
     // Groups that were passed on the command line
-    for (String group : groups) {
-      result.put(group, group);
-    }
+    Map<String, String> result = new HashSet<>(groups).stream().collect(Collectors.toMap(g ->g, g->g));
```
Comment (juherr) [inline src/main/java/org/testng/internal/GroupsHelper.java]: Style (space before/after ->)

File: src/main/java/org/testng/internal/MethodGroupsHelper.java [hunk 2]
```diff
@@ -162,15 +163,9 @@ protected static void findGroupTransitiveClosure(
...
-    for (ITestNGMethod m : includedMethods) {
-      runningMethods.put(m, m);
-    }
+    Map<ITestNGMethod, ITestNGMethod> runningMethods = includedMethods.stream().collect(Collectors.toMap(m -> m, m -> m));
 
-    Map<String, String> runningGroups = Maps.newHashMap();
-    for (String thisGroup : includedGroups) {
-      runningGroups.put(thisGroup, thisGroup);
-    }
+    Map<String, String> runningGroups = Arrays.stream(includedGroups).collect(Collectors.toMap(g -> g, g->g));
```
Comment (juherr) [inline src/main/java/org/testng/internal/MethodGroupsHelper.java]: style

File: src/test/java/test/override/OverrideTest.java [hunk 3]
```diff
@@ -1,30 +1,40 @@
...
  * Verify that command line switches override parameters in testng.xml.
  *
  * @author Cedric Beust <cedric@beust.com>
  */
 public class OverrideTest extends SimpleBaseTest {
 
+  private static File createTempFile(String content) {
+    try {
+      // Create temp file.
+      File result = File.createTempFile("testng-tmp", "");
```
Comment (juherr) [inline src/test/java/test/override/OverrideTest.java]: Prefer to use `Path`

**Review by juherr — APPROVED** [review]

File: src/main/java/org/testng/internal/GroupsHelper.java (hunk 1 above)
Comment (krmahadevan) [reply src/main/java/org/testng/internal/GroupsHelper.java]: Fixed.

File: src/main/java/org/testng/internal/MethodGroupsHelper.java (hunk 2 above)
Comment (krmahadevan) [reply src/main/java/org/testng/internal/MethodGroupsHelper.java]: fixed

**Pull request merged**